packages/styled-components/src/utils/flatten.js

```diff
@@ -55,12 +56,13 @@ export default function flatten(chunk: any, executionContext: ?Object, styleShee
 
   /* Either execute or defer the function */
   if (isFunction(chunk)) {
-    if (executionContext) {
+    if (isStatelessFunction(chunk) && executionContext) {
```

Comment:
Since classes used to go into this code path and be checked if they are styled components or not, is this a breaking change?

Reply:
The `isStyledComponent` check happens higher in the function, so shouldn't be.